gateway/api/views.py

```diff
@@ -187,6 +190,67 @@ def stop(self, request, pk=None):  # pylint: disable=invalid-name,unused-argumen
         return Response({"message": message})
 
 
+class FilesViewSet(viewsets.ViewSet):
+    """ViewSet for file operations handling.
+
+    Note: only tar files are available for list and download
+    """
+
+    BASE_NAME = "files"
+
+    # @action(methods=["GET"], detail=False)
+    def list(self, request):
+        """List of available for user files."""
+        files = []
+        tracer = trace.get_tracer("gateway.tracer")
+        ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
+        with tracer.start_as_current_span("gateway.files.list", context=ctx):
+            user_dir = os.path.join(settings.MEDIA_ROOT, request.user.username)
+            if os.path.exists(user_dir):
+                files = [
+                    os.path.basename(path) for path in glob.glob(f"{user_dir}/*.tar")
+                ]
+            else:
+                logger.warning(
+                    "Directory %s does not exist for %s.", user_dir, request.user
+                )
+
+        return Response({"results": files})
+
+    @action(methods=["GET"], detail=False)
+    def download(self, request):  # pylint: disable=invalid-name
+        """Download selected file."""
+        response = Response(
```

Comment:
nit
```suggestion
        # default response for file not found, overwritten if file is found
        response = Response(
```
or something like that